src/main.go

```diff
@@ -3,19 +3,41 @@ package main
 import (
 	"io/ioutil"
     "log"
+	"os"
 	"fmt"
 	game "game"
 )
 
 func main() {
+	fileName := os.Args[1]
 
-    content, err := ioutil.ReadFile("./src/datasets/empty_board.txt")
+	fmt.Println(fileName)
+    
+	content, err := ioutil.ReadFile("./src/datasets/" + fileName)
 
-     if err != nil {
-          log.Fatal(err)
-     }
-	game.DeserializeBoard(string(content));
+	if err != nil {
+		log.Fatal(err)
+	}
 
-	fmt.Println("Deserialization has been made with success :")
 	fmt.Println(string(content))
+	board, err := game.DeserializeBoard(string(content))
```

Comment:
So I assume the internal representation of a game remains an array? That means you'll have to do the conversion each time you want to test for victory?

Reply:
I think I must have an array based representation of the game to handle thousands of quarter's rotations.
If I use binary numbers, I should convert it to an array, rotate it, and convert it again into binary number to detect victory.

But I think it's better if I have an array of 4 quarters.

Reply:
If you keep each quarter as a separate number, you could use the bitwise operator shift left and shift right (`<<` and `>>`) to implement quarter rotation. But that's probably overkill. 

(if you want to go down the rabbit hole, rotating a 3x3 binary matrix can be done using 24 binary operations - `&`, `>>` and `|`. Probably super fast but absolutely not legible. see https://stackoverflow.com/a/41046873/1333479 for an idea of what I mean)